src/relo_core.c

```diff
@@ -1732,7 +1847,7 @@ int bpf_core_apply_relo_insn(const char *prog_name, struct bpf_insn *insn,
 	}
 
 patch_insn:
-	if (reloc_info) {
+	if (reloc_info && targ_res.targ_spec) {
```

Comment:
Did you hit any case where `targ_res.targ_spec` was null? I'd like to know when this happens.

Reply:
I did. For local relocations. Check in the same function, in the beginning. There is a jump to patch_insn.